lib/src/signed_video_h26x_common.c

```diff
@@ -874,6 +874,12 @@ check_and_copy_hash_to_hash_list(signed_video_t *self, const uint8_t *hash, size
   openssl_update_hash(self->crypto_handle, hash, hash_size, true);
 }
 
+/*
+ * Updates the |linked_hash| buffer with the upcoming hash. The buffer contains 2 slots for hashes.
+ * When a new I NALU is encountered in the stream, the hash of the |linked_hash| buffer is updated
```

Comment:
There is nothing in this function that says it only applies the update on I-frames only. Remove this sentence.